packages/opentelemetry-node/src/NodeTracerFactory.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NodeTracer } from './NodeTracer';
+import { NodeTracerConfig } from './config';
+import { BasicTracerFactory } from '@opentelemetry/tracing';
+
+export class NodeTracerFactory extends BasicTracerFactory {
```

Comment:
Why does this inherit `BasicTracerFactory` rather than `AbstractTracerFactory<NodeTracer>>`?

Reply:
> I don't see the reason why we would reimplement getTracer to do the addSpanProcessor loops, the only thing that's different is _newTracer. That being said I don't really have a strong opinion here, can go ahead and make the change